libcrux-ml-dsa/src/samplex4.rs

```diff
@@ -10,6 +10,18 @@ fn generate_domain_separator(row: u8, column: u8) -> u16 {
     (column as u16) | ((row as u16) << 8)
 }
 
+type Matrix<SIMDUnit, const ROWS_IN_A: usize, const COLUMNS_IN_A: usize> =
+    [[PolynomialRingElement<SIMDUnit>; COLUMNS_IN_A]; ROWS_IN_A];
+
+fn update_matrix<SIMDUnit: Operations, const ROWS_IN_A: usize, const COLUMNS_IN_A: usize>(
```

Comment:
Ugh, that's pretty ugly. If this is really needed, it needs an issue and some coments here.

Reply:
Yes it is unfortunate but it is because F* does not seem to deal well with sequences of deep updates.
I added an issue and am trying to reproduce for the F* team: https://github.com/hacspec/hax/issues/1098